.phpcs.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress Coding Standards">
+	<description>Apply WordPress Coding Standards</description>
+
+	<!-- Set the memory limit to 256M.
+		 For most standard PHP configurations, this means the memory limit will temporarily be raised.
+		 Ref: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#specifying-phpini-settings
+	-->
+	<ini name="memory_limit" value="256M"/>
+
+	<!-- Whenever possible, cache the scan results and re-use those for unchanged files on the next scan. -->
+	<arg name="cache"/>
+
+	<!-- Strip the filepaths down to the relevant bit. -->
+	<arg name="basepath" value="./"/>
+
+	<!-- Check up to 20 files simultaneously. -->
+	<arg name="parallel" value="20"/>
+
+	<!-- Show sniff codes in all reports. -->
+	<arg value="ps"/>
+
+	<!-- Use WordPress Coding Standards. -->
+	<rule ref="WordPress-Extra">
+		<!-- Sprinkle in WebDevStudios specifc rules and excludes. -->
+		<exclude name="Generic.Arrays.DisallowShortArraySyntax" />
+	</rule>
+	<rule ref="WordPress-Docs" />
+
+	<!-- Use WordPress PHP Compatibility. -->
+	<rule ref="PHPCompatibilityWP"/>
+
+	<!-- WordPress Core currently supports PHP 5.6+. -->
+	<config name="testVersion" value="5.6-"/>
+
+	<!-- Only sniff PHP files. -->
+	<arg name="extensions" value="php"/>
+
+	<!-- Only sniff the theme. -->
+	<file>./</file>
+
```

Comment:
I think we should include a few more "helpful" rules, like allowing file name exceptions for themes, defining the text domain and function prefixes. 

The customizable WPCS sniffs are all listed in the [WPCS wiki](https://github.com/WordPress/WordPress-Coding-Standards/wiki/Customizable-sniff-properties).

```
<!-- Allow for theme specific exceptions to the file name rules based
	 on the theme hierarchy. -->
<rule ref="WordPress.Files.FileName">
	<properties>
		<property name="is_theme" value="true"/>
	</properties>
</rule>

<!-- Verify that everything in the global namespace is prefixed with a theme specific prefix.
	 Multiple valid prefixes can be provided as a comma-delimited list. -->
<rule ref="WordPress.NamingConventions.PrefixAllGlobals">
	<properties>
		<property name="prefixes" type="array" value="_s" />
	</properties>
</rule>

<!-- Verify that the text_domain is set to the desired text-domain.
	 Multiple valid text domains can be provided as a comma-delimited list. -->
<rule ref="WordPress.WP.I18n">
	<properties>
		<property name="text_domain" type="array" value="_s"/>
	</properties>
</rule>

```

Reply:
Maybe set the minimum supported WP version too.
Looks like our style.css says WP 5.5
```
	<!-- Set the minimum supported WP version. This is used by several sniffs. -->
	<config name="minimum_supported_wp_version" value="5.5"/>
```

Reply:
Thanks, @richaber 

During my first pass at testing I get:

```text
The "_s" prefix is too short. Short prefixes are not unique enough and may cause name collisions with other code.
```

Due to this rule:

```xml
<!-- Verify that everything in the global namespace is prefixed with a theme specific prefix.
	 Multiple valid prefixes can be provided as a comma-delimited list. -->
<rule ref="WordPress.NamingConventions.PrefixAllGlobals">
	<properties>
		<property name="prefixes" type="array" value="_s" />
	</properties>
</rule>
```

Thoughts here?

Reply:
```
<rule ref="WordPress">
	<exclude name="WordPress.NamingConventions.PrefixAllGlobals.ShortPrefixPassed"/>
</rule>
```

Reply:
@richaber Got it! See https://github.com/WebDevStudios/wd_s/pull/618/commits/8c1b9861e3e7ef4c9d444846160960275f935669

Thank you, sir!